src/types/currency.ts

```diff
@@ -69,36 +83,18 @@ function bytesFromRepresentation(input: string): Buffer {
 class Currency extends Hash160 {
   static readonly XRP = new Currency(Buffer.alloc(20));
   private readonly _iso?: string;
-  private readonly _isNative: boolean;
 
   constructor(byteBuf: Buffer) {
     super(byteBuf ?? Currency.XRP.bytes);
-
-    let onlyISO = true;
-
-    const bytes = this.bytes;
     const code = this.bytes.slice(12, 15);
-    const iso = code.toString();
 
-    for (let i = bytes.length - 1; i >= 0; i--) {
-      if (bytes[i] !== 0 && !(i === 12 || i === 13 || i === 14)) {
-        onlyISO = false;
-        break;
-      }
+    if (this.bytes[0] !== 0 && this.bytes[1] !== 0) {
```

Comment:
This is what the spec says is correct: https://xrpl.org/serialization.html#currency-codes

Reply:
According to that documentation, wouldn't it only be `this.bytes[0]` that has to be non-zero to make this a non-ISO currency???

Reply:
yup agreed with nathan, good call dude!

Reply:
Fixed